samples/performance/constant_data/constant_data_tutorial.md

```diff
@@ -0,0 +1,399 @@
+
+However the Vulkan API gives a lot of flexibility about how to handle **descriptor sets**, offering many different types and different ways to bind and use them (especially when we factor in extensions). This can puzzle developers about which is best, and for which scenarios. This tutorial aims to ease some of the confusion and uncertainty around this subject. 
+
+When we break this down, we have the following methods:
+
+* Push Constants
+* Descriptor Sets
+* Dynamic Descriptor Sets
+* Update-after-bind Descriptor Sets
+* Buffer array with dynamic indexing
+* [Inline uniform buffer objects](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_EXT_inline_uniform_block.html) (click to read more)
+* [Push descriptors](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_KHR_push_descriptor.html) (click to read more)
+
+**Inline uniform buffer objects** and **push descriptors** are not covered by this tutorial, please use the links above to learn more about them.
+
+## **Sample Overview**
+
+### **Introduction**
+
+The sample uses a mesh heavy scene which has 1856 meshes with 475 KB of mesh data. This is to demonstrate a use case where many different calls to pushing uniform data will occur during a single frame. The constant data that is being sent is the per-mesh model matrix, the camera view projection matrix, and if 256 bytes of push constants are available it will push a scale matrix and some extra padding.
+
+A performance graph is displayed at the top with two charts, one showing frame time, and one showing the load/store cycles.
+
+![graph](images/graph_data.jpg)
+
+These two counters will show the CPU and GPU cost respectively, so when you go to toggle the different method you can see how it changes.
+
+### **Controls**
+
+The options presented to the user lets them change the method by which we push the MVP data.
+
+![sample](images/controls.png)
+
+It is important to note that the configuration adapts to the device and GPU. This is so if an extension isn't supported, the related option will no longer show. You can check the console log to see a warning message detailing what features were disabled and why.
+
+When an option is changed, the descriptor sets are flushed and recreated with their new setup, and the respective render pipeline/subpass.
+
+## **Push Constants**
+
+### **Introduction**
+
+Push constants are usually the first method newer Vulkan programmers will stumble upon when beginning to work with constant data. They are straightforward to use and integrate nicely into any codebase, making them a great option to send simple data to your shaders.
+
+The downside to push constants is that they have very strict limitations on how much data can be sent.
+The Vulkan spec requires drivers to support at least 128 bytes of push constants, and most modern implementations of Vulkan will support 256 bytes. 
```

Comment:
or much more. It is important to query the limits for the device to determine the sensible choice at runtime.

Reply:
support **at least** 256 bytes, **sometimes much more.**